src/rebar_dialyzer.erl

```diff
@@ -0,0 +1,235 @@
+-module(rebar_dialyzer).
+
+-export([
+         dialyze/2,
+         'build-plt'/2,
+         'check-plt'/2,
+         'delete-plt'/2
+        ]).
+
+%% for internal use only
+-export([info/2]).
+
+-include("rebar.hrl").
+
+%% ===================================================================
+%% Public API
+%% ===================================================================
+
+%% TODO:
+%%
+%%  * How can we configure the filtering of warnings without exact line
+%%    numbers as a way to avoid updating the config whenever the file
+%%    in question is modified (and a warning's line number changes)?
+%%    Maybe use '_' for line number in dialyzer_ignored_warnings or a
+%%    separate dialyzer_ignored_warnings_noline (name to be determined)
+%%    list, and subtract that from the dialyzer:run/2 result.
+
+dialyze(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ok = check_plt_existence(Plt),
+
+    Opts = [
+            {analysis_type, succ_typings},
+            {init_plt, Plt},
+            {files_rec, ["ebin"]},
+            {warnings, warnings(Config)}
+           ],
+    ?DEBUG("dialyze opts:~n~p~n", [Opts]),
+    RawWs = dialyzer:run(Opts),
+    case filter_warnings(RawWs, Config) of
+        [] ->
+            {ok, NewConfig};
+        Ws ->
+            print_warnings(Ws),
+            ?FAIL
+    end.
+
+'build-plt'(Config, AppFile) ->
+    {Config1, AppDirs} = app_dirs(Config, AppFile),
+    {NewConfig, Plt} = plt(Config1, AppFile),
+    Opts = [
+            {analysis_type, plt_build},
+            {output_plt, Plt},
+            {files_rec, AppDirs}
+           ],
+    ?DEBUG("build-plt opts:~n~p~n", [Opts]),
+    case dialyzer:run(Opts) of
+        [] ->
+            {ok, NewConfig};
+        Ws ->
+            %% As plt_build may raise warnings but still successfully
+            %% create the PLT, we cannot interpret this as failure,
+            %% and therefore all we can do is report warnings.
+            print_warnings(Ws)
+    end.
+
+'check-plt'(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ok = check_plt_existence(Plt),
+
+    Opts = [
+            {analysis_type, plt_check},
+            {init_plt, Plt}
+           ],
+    ?DEBUG("build-plt opts:~n~p~n", [Opts]),
+    case dialyzer:run(Opts) of
+        [] ->
+            {ok, NewConfig};
+        Ws ->
+            print_warnings(Ws),
+            ?FAIL
+    end.
+
+'delete-plt'(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ?DEBUG("Delete PLT '~s'~n", [Plt]),
+    ok = rebar_file_utils:delete_each([Plt]),
+    {ok, NewConfig}.
+
+%% ===================================================================
+%% Internal functions
+%% ===================================================================
+
+info(help, dialyze) ->
+    info_help("Analyze the code for discrepancies");
+info(help, 'build-plt') ->
+    info_help("Build project-specific PLT");
```

Comment:
Correct, and the idea is that the included apps change seldom enough that sharing or at least preserving the generated PLT will save time. This does of course assume the project's apps do not change frequently. If that's not the case, a local plt in `<base_dir>/.rebar/` is more useful, and you'll have to remember to run `build-plt` whenever apps change.